### ISSUE TITLE
Sometimes retrofit does not seem to close the ResponseBody

### ISSUE THREAD

**Issue by UweTrottmann** [UweTrottmann/SeriesGuide#561]
Do we actually need to make sure to close this ourselves? Judging from the [OkHttpCall.parseResponse() implementation](https://github.com/square/retrofit/blob/parent-2.4.0/retrofit/src/main/java/retrofit2/OkHttpCall.java#L191) either there or in (at least) the (Gson) converter the body is closed. But apparently not in all cases.

Retrofit 2.3.0.

- [x] Test with `2.4.0`.
- [x] Investigate.
```
2018-07-20 07:39:16.415 6160-6275/com.battlelancer.seriesguide D/StrictMode: StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
    at android.os.StrictMode$AndroidCloseGuardReporter.report(StrictMode.java:1786)
    at dalvik.system.CloseGuard.warnIfOpen(CloseGuard.java:264)
    at java.lang.reflect.Method.invoke(Native Method)
    at okhttp3.internal.platform.AndroidPlatform$CloseGuard.warnIfOpen(AndroidPlatform.java:353)
    at okhttp3.internal.platform.AndroidPlatform.logCloseableLeak(AndroidPlatform.java:160)
    at okhttp3.ConnectionPool.pruneAndGetAllocationCount(ConnectionPool.java:271)
    at okhttp3.ConnectionPool.cleanup(ConnectionPool.java:211)
    at okhttp3.ConnectionPool$1.run(ConnectionPool.java:60)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:764)
 Caused by: java.lang.Throwable: Explicit termination method 'response.body().close()' not called
    at dalvik.system.CloseGuard.open(CloseGuard.java:221)
    at java.lang.reflect.Method.invoke(Native Method)
    at okhttp3.internal.platform.AndroidPlatform$CloseGuard.createAndOpen(AndroidPlatform.java:341)
    at okhttp3.internal.platform.AndroidPlatform.getStackTraceForCloseable(AndroidPlatform.java:156)
    at okhttp3.RealCall.captureCallStackTrace(RealCall.java:89)
    at okhttp3.RealCall.execute(RealCall.java:73)
    at retrofit2.OkHttpCall.execute(OkHttpCall.java:180)
    at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall.execute(ExecutorCallAdapterFactory.java:91)
    at com.battlelancer.seriesguide.ui.movies.MovieTools.getMovieSummary(MovieTools.java:529)
    at com.battlelancer.seriesguide.ui.movies.MovieTools.loadSummaryFromTmdb(MovieTools.java:497)
    at com.battlelancer.seriesguide.ui.movies.MovieTools.getMovieDetails(MovieTools.java:473)
    at com.battlelancer.seriesguide.ui.movies.MovieTools.addMovies(MovieTools.java:419)
    at com.battlelancer.seriesguide.sync.TraktMovieSync.syncLists(TraktMovieSync.java:197)
    at com.battlelancer.seriesguide.sync.TraktSync.sync(TraktSync.java:101)
    at com.battlelancer.seriesguide.sync.SgSyncAdapter.onPerformSync(SgSyncAdapter.java:161)
    at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:334)
```

**Comment by JakeWharton** [comment]
Nothing has changed with Retrofit here between 2.3.0 and 2.4.0 that I can
tell. It might be a leak in OkHttp when served from the cache. While you
should still upgrade to 2.4.0, I would also consider forcing OkHttp to the
recently-released 3.11.0 to see if it still happens.

**Comment by UweTrottmann** [comment]
@JakeWharton Thanks for confirming that 2.4.0 has no relevant changes.

Still happening if using 2.4.0 and OkHttp 3.11.0.

Test 1: fresh install, connect to trakt with test account, observe errors when movies are auto-added. Might be a coincidence, but I am seeing more of those on an Android P emulator vs. Android O_MR1.

Test 2: add a bunch of shows. Clean cache. Do Sync & Re-download all.
Not seeing those then.

So less likely an issue due to running from SyncAdapter (both tests do).

So maybe this is an issue with the tmdb-java lib (might be, there were some large changes to the authenticator and interceptor I probably should not have accepted...).
